packages/react-components/components/WithPasteAware.tsx

```diff
@@ -0,0 +1,94 @@
+// HOC to add a paste button to a text input
+
+import * as React from 'react'
+import { AppState, Clipboard, TextInputProps } from 'react-native'
+
+interface PasteAwareProps {
+  value: string
+  shouldShowClipboard: (value: string) => boolean
+  onChangeText: (text: string) => void
+}
+
+export interface PasteAwareWrappedElementProps {
+  isPasteIconVisible: boolean
+  onPressPaste: () => void
+  onChangeText: (text: string) => void
+}
+
+interface PasteAwareState {
+  isPasteIconVisible: boolean
+  clipboardContent: string | null
+}
+
+export function withPasteAware<P extends TextInputProps>(
+  WrappedView: React.ComponentType<P & PasteAwareWrappedElementProps>
+) {
+  return class WithPasteAware extends React.Component<P & PasteAwareProps> {
+    state: PasteAwareState = {
+      isPasteIconVisible: false,
+      clipboardContent: null,
+    }
+
+    _interval?: number
+    _isMounted = false
+
+    async componentDidMount() {
+      this._isMounted = true
+      AppState.addEventListener('change', this.checkClipboardContents)
+      this._interval = window.setInterval(async () => {
+        await this.checkClipboardContents()
+      }, 1000) // Every 1s
+      await this.checkClipboardContents()
```

Comment:
This strategy (which I realize is based partly on my own code) may not be viable anymore for the new iOS versions. 
https://9to5mac.com/2020/06/24/ios-14-clipboard-notifications/

Reply:
Given that this isn't really new and we do it in other places (verification), I'm okay with just leaving a TODO to address it later